### PR TITLE
[python] Enhancement of manual test using Cirque

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem & Change overview

This PR is an enhancement of existing manual testing using cirque, with the following features:
- Mount CHIP checkout directory to container
- Add example topologies

#### Testing
- If manually tested, what platforms controller and device platforms were manually tested, and how?
  Tested on linux workstation

### CODE REVIEW

**Comment by yunhanw-google** [comment]
pleaes update md file as well, thanks

File: integrations/docker/ci-only-images/chip-cirque-device-base/Dockerfile [hunk 1]
```diff
@@ -15,6 +15,7 @@ WORKDIR /app
 
 RUN apt-get update \
   && apt-get install --no-install-recommends -y sudo git ca-certificates psmisc dhcpcd5 wpasupplicant wireless-tools \
+                                                gdb python3 python3-pip libcairo2-dev libjpeg-dev libgif-dev python3-dev \
```
Comment (doublemis1) [inline integrations/docker/ci-only-images/chip-cirque-device-base/Dockerfile:18]: You can add avahi-utils package if this image will be used to simulate the controller.
This package might help with verifying if the discoverable node service is propagated using `avahi-browse -rt _chipc._udp` command.

Comment (erjiaqing) [reply integrations/docker/ci-only-images/chip-cirque-device-base/Dockerfile:18]: Added, thanks.

**Review by yunhanw-google — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

File: src/test_driver/linux-cirque/README.md [hunk 2]
```diff
@@ -10,6 +10,18 @@ In Project CHIP, cirque is used for integration tests.
 There is a script for running cirque tests, you can find it at
 `scripts/tests/cirque_tests.sh`
 
+## "Docker out of docker" setup
+
+If you don't want cirque break your local environment, you can run
+
+```
+integrations/docker/images/chip-build-cirque/run.sh bash
```
Comment (yunhanw-google) [inline src/test_driver/linux-cirque/README.md:18]: this does not work, please consider ./integrations/docker/images/chip-build-cirque/run.sh --privileged --env LOG_DIR=/tmp/cirque_test_output --env CLEANUP_DOCKER_FOR_CI=1 --volume /dev/pts:/dev/pts --volume /tmp/cirque_test_output:/tmp/cirque_test_output -it -- bash

**Review by yunhanw-google — CHANGES_REQUESTED** [review]
on hold

File: src/test_driver/linux-cirque/README.md (hunk 2 above)
Comment (erjiaqing) [reply src/test_driver/linux-cirque/README.md]: Adjusted some arguments for local run.

**Review by yunhanw-google — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**